apis/nucleus/src/components/Sheet.jsx

```diff
@@ -47,115 +51,141 @@ function getBounds(pos, columns, rows) {
         */
 
-        // Clear the cell list
-        cells.forEach((c) => {
-          root.removeCell(c.currentId);
-        });
+          // Clear the cell list
+          cells.forEach((c) => {
+            root.removeCell(c.currentId);
+          });
 
-        const lCells = layout.cells;
-        const { columns, rows } = layout;
-        // TODO - should try reuse existing objects on subsequent renders
-        // Non-id updates should only change the "css"
-        const cs = await Promise.all(
-          lCells.map(async (c) => {
-            let mounted;
-            const mountedPromise = new Promise((resolve) => {
-              mounted = resolve;
-            });
+          const lCells = layout.cells;
+          const { columns, rows } = layout;
+          // TODO - should try reuse existing objects on subsequent renders
+          // Non-id updates should only change the "css"
+          const cs = await Promise.all(
+            lCells.map(async (c) => {
+              let mounted;
+              const mountedPromise = new Promise((resolve) => {
+                mounted = resolve;
+              });
 
-            const cell = cells.find((ce) => ce.id === c.name);
-            if (cell) {
-              cell.bounds = getBounds(c, columns, rows);
-              delete cell.mountedPromise;
-              return cell;
-            }
-            const vs = await getObject({ id: c.name }, halo, modelStore);
-            return {
-              model: vs.model,
-              id: c.name,
-              bounds: getBounds(c, columns, rows),
-              cellRef: React.createRef(),
-              currentId: uid(),
-              mounted,
-              mountedPromise,
-            };
-          })
-        );
-        cs.forEach((c) => root.addCell(c.currentId, c.cellRef));
-        setCells(cs);
+              const cell = cells.find((ce) => ce.id === c.name);
+              if (cell) {
+                cell.bounds = getBounds(c, columns, rows);
+                delete cell.mountedPromise;
+                return cell;
+              }
+              const vs = await getObject({ id: c.name }, halo, modelStore);
+              return {
+                model: vs.model,
+                id: c.name,
+                bounds: getBounds(c, columns, rows),
+                cellRef: React.createRef(),
+                currentId: uid(),
+                mounted,
+                mountedPromise,
+              };
+            })
+          );
+          cs.forEach((c) => root.addCell(c.currentId, c.cellRef));
+          setCells(cs);
+        };
+        fetchObjects();
+      }
+    }, [layout]);
+
+    useEffect(() => {
+      const onModelClose = () => {
+        model.removeListener('closed', onModelClose);
+        unmount();
       };
-      fetchObjects();
-    }
-  }, [layout]);
+      model.on('closed', onModelClose);
+      return () => model.removeListener('closed', onModelClose);
+    }, [model]);
 
```

Comment:
One last thing, I would like to move this code out into sheet.js. As that is the "object" that owns this component and should be responsible for unmounting. However that requires a bit of restructuring as the new model is never passed through that sheet API, like changing the navigations setSheetRef to setSheetAPI and add a set model on the sheet API (could be useful for other purposes actually).
So not gonna block you with this, just consider it. I think the current setup works just fine, and is all internal anyways.